addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -809,7 +809,8 @@ odoo.define('pos_coupon.pos', function (require) {
 
             // Check if valid customer
             const customer = this.get_client();
-            if (program.rule_partners_domain && !program.valid_partner_ids.has(customer ? customer.id : 0)) {
+            const partnersDomain = program.rule_partners_domain || '[]';
+            if (partnersDomain !== '[]' && !program.valid_partner_ids.has(customer ? customer.id : 0)) {
```

Comment:
```suggestion
            if (program.valid_partner_ids.has(customer ? customer.id : 0)) {
```
Can you try this instead?

Reply:
In this case, the coupon can not be applied if they don't select a customer. This was the main problem.

Reply:
True true.